File: computer_programming/assignment_01_formula_translation/code/09.cpp
# include <iostream>

// For calculating the square root of number we need to use the function sqrt() which has been defined in math.h header file.
# include <math.h>

using namespace std;

int main()
{
	// Define the variables for storing the coordinates of two points
	int x1, y1, x2, y2;

	// Take the coordinates of first point from the user and assign them to our variables
	cout << "Please enter the coordinates of Point-1 \n( input format: x y ): ";
	cin >> x1 >> y1;

	// Now take the coordinates of the second point and assign them to our variables
	cout << "\nPlease enter the coordinates of Point-2 \n( input format: x y ): ";
	cin >> x2 >> y2;

	// Note: The formula for calculating distance between 2 point coordinates is :
	// [ (x2 - x1)^2 + (y2 - y1)^2 ]^1/2

	// we will need some more variables during runtime to simplify things
	// h_dist = horizontal distance = change in x between points
	// v_dist = vertical distance = change in y between points
	float h_dist, v_dist;

	h_dist = x2 - x1;
	v_dist = y2 - y1;

	// Let's calculate the answer and neatly display it to user
	cout << "\nThe distance between points is: " << sqrt( (h_dist*h_dist) + (v_dist*v_dist) ) << endl;

	// end program
	return 0;
}

File: computer_programming/01_formula_translation/code/python/09.py
x1, y1 = input("Enter coordinates of Point 1: ").split()
x2, y2 = input("Enter coordinates of Point 2: ").split()

x1 = int(x1)
y1 = int(y1)
x2 = int(x2)
y2 = int(y2)

xDistance = x2 - x1
yDistance = y2 - y1

result = ( xDistance**2 + yDistance**2 ) ** 0.5
print("Distance between points is %.2f" % result)
